src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3371,23 +3371,41 @@ s32 Player_SetupAction(PlayState* play, Player* this, PlayerActionFunc actionFun
     return 1;
 }
 
-void func_80835DAC(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
-    s32 temp;
+/**
+ * Calls `Player_SetupAction` to setup a new action, but takes extra measures to
+ * preserve AnimMovement while doing so.
+ */
+void Player_SetupActionPreserveAnimMovement(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
+    s32 savedMovementFlags;
 
-    temp = this->skelAnime.movementFlags;
+    savedMovementFlags = this->skelAnime.movementFlags;
+
+    // Setting `skelAnime.movementFlags` to 0 will prevent `Player_FinishAnimMovement` from ending
+    // AnimMovement when `Player_SetupAction` is called.
     this->skelAnime.movementFlags = 0;
+
     Player_SetupAction(play, this, actionFunc, flags);
-    this->skelAnime.movementFlags = temp;
+    this->skelAnime.movementFlags = savedMovementFlags;
 }
 
-void func_80835DE4(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
-    s32 temp;
+/**
+ * Calls `Player_SetupAction` to setup a new action, but takes extra measures to
+ * preserve the current itemAction while doing so.
+ *
+ * Note that `itemAction` must be PLAYER_IA_NONE or higher for the action change to take place.
```

Comment:
should the be promoted to be in the lead paragraph instead of as a side note? It seems important, what would happen in the rest of this code if the check was removed?

Reply:
When I say "note" im not trying to say its a side note/not important. If anything its bringing attention to it.

As for why, Im not entirely sure. The -1 item action means player is currently shielding, but I dont know too much about it beyond that. I think its a temp value to mean that it should still preserve the item in hand so it can come back after shielding is over. But yeah I dont know how it plays into this.